src/objects/BatchedMesh.js

```diff
@@ -344,23 +345,36 @@ class BatchedMesh extends Mesh {
 
 	addInstance( geometryId ) {
 
+		const atCapacity = this._drawInfo.length >= this.maxInstanceCount;
+
 		// ensure we're not over geometry
-		if ( this._drawInfo.length >= this._maxInstanceCount ) {
+		if ( atCapacity && this._fragInfo.length === 0 ) {
 
 			throw new Error( 'BatchedMesh: Maximum item count reached.' );
 
 		}
 
-		this._drawInfo.push( {
-
+		const instanceDrawInfo = {
 			visible: true,
 			active: true,
 			geometryIndex: geometryId,
+		};
 
-		} );
+		let drawId = null;
+
+		// If at capacity, rewrite over an inactive block
+		if ( atCapacity ) {
```

Comment:
I think we should check if there are any available deleted Ids and reuse those ids first rather than waiting until the full capacity has been used.

Reply:
I like this idea. It would be best to keep the available deleted ids as small as possible. I will make that change